utils_cv/tracking/model.py

```diff
@@ -243,9 +279,48 @@ def save(self, path) -> None:
         save_model(path, self.epoch, self.model, self.optimizer)
         print(f"Model saved to {path}")
 
-    def evaluate(self, results, gt) -> pd.DataFrame:
-        pass
+    def evaluate(self,
+                 results: Dict[int, List[TrackingBbox]],
+                 gt_root_path: str) -> pd.DataFrame:
+        
+        """ eval code that calls on 'motmetrics' package in referenced FairMOT script, to produce MOT metrics on inference, given ground-truth.
+        Args:
+            results: prediction results from predict() function, i.e. Dict[int, List[TrackingBbox]] 
+            gt_root_path: path of dataset containing GT annotations in MOTchallenge format (xywh)
+        Returns:
+            strsummary: pandas.DataFrame output by method in 'motmetrics', containing metrics scores
+        Raises:
+            Exception: if both `results` and `self.stored_predictions` are None.
+        """
+        if results is None:
```

Comment:
What is the purpose of "self.stored_predictions" and can it get deleted? Then this section "if results" would not be necessary anymore.

Reply:
Good point, i was doing it like in detection/model.py. But the context is different because it could use the predictions from test dataset stored internally. Will remove